internal/dataplane/kongstate/kongstate.go

```diff
@@ -356,3 +356,30 @@ func globalPlugins(log logrus.FieldLogger, s store.Storer) ([]Plugin, error) {
 func (ks *KongState) FillPlugins(log logrus.FieldLogger, s store.Storer) {
 	ks.Plugins = buildPlugins(log, s, ks.getPluginRelations())
 }
+
+// FillIDs iterates over the KongState and fills in the ID field for each entity
+// that supports the FillID method (these are Service, Route and Consumer). It makes
+// their IDs deterministic, enabling their correct identification in external systems
+// (e.g. Konnect Analytics).
+func (ks *KongState) FillIDs(logger logrus.FieldLogger) {
+	for svcIndex, svc := range ks.Services {
+		if err := svc.FillID(); err != nil {
+			logger.WithError(err).Errorf("failed to fill ID for service %s", *svc.Name)
+		}
+		ks.Services[svcIndex] = svc
```

Comment:
Just a small nit: all those 3 entities are saved regardless of `err != nil` condition. Is that the expectation? I believe this might be surprising to the reader.

Reply:
Yeah, it might make sense to be explicit here and assign only when `err == nil`. Right now the implementations of `FillID` do not modify entities, but who knows what the future will bring. Let's be future-proof. 👍 PTAL